src/libraries/System.Private.CoreLib/src/System/Guid.Windows.cs

```diff
@@ -5,22 +5,26 @@ namespace System
 {
     public partial struct Guid
     {
-        public static Guid NewGuid()
+        public static unsafe Guid NewGuid()
         {
-            // CoCreateGuid should never return Guid.Empty, since it attempts to maintain some
-            // uniqueness guarantees.
+            // CoCreateGuid should never return Guid.Empty, since it attempts to maintain some uniqueness guarantees.
 
-            int hr = Interop.Ole32.CoCreateGuid(out Guid g);
-            // We don't expect that this will ever throw an error, none are even documented, and so we don't want to pull
-            // in the HR to ComException mappings into the core library just for this so we will try a generic exception if
-            // we ever hit this condition.
+            Guid g;
+            int hr = Interop.Ole32.CoCreateGuid(&g);
             if (hr != 0)
             {
-                Exception ex = new Exception();
-                ex.HResult = hr;
-                throw ex;
+                ThrowForHr(hr);
             }
+
             return g;
         }
+
+        private static void ThrowForHr(int hr)
```

Comment:
Just a quick question: is NoInlining needed on throw helpers like this one nowadays?

Reply:
no

Reply:
@En3Tho in fact adding `NoInlining` produces worse code, because the JIT cannot see that this method is a throw helper, which would mark it as a rarely executed block and place it away from the main code.